### ISSUE TITLE
Fix -Wstringop-overflow warnings in fs/smb/client/cifssmb.c

### ISSUE THREAD

**Issue by GustavoARSilva** [KSPP/linux#310]
Found in powerpc64 and s390 (and a couple more archs).
```c
fs/smb/client/cifssmb.c:2987:31: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
fs/smb/client/cifssmb.c:2988:30: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
```

**Comment by GustavoARSilva** [comment]
Patch: https://lore.kernel.org/linux-hardening/ZK3h3+dHBGONHt+S@work/

**Comment by GustavoARSilva** [comment]
fixed in commit f1f047bd7ce0d73788e04ac02268060a565f7ecb
